flake8_async/visitors/visitor102.py

```diff
@@ -60,7 +64,11 @@ def async_call_checker(
         if self._critical_scope is not None and not any(
             cm.has_timeout and cm.shielded for cm in self._trio_context_managers
         ):
-            self.error(node, self._critical_scope)
+            if self._critical_scope.name == "except":
+                error_code = "ASYNC120"
+            else:
+                error_code = "ASYNC102"
+            self.error(node, self._critical_scope, error_code=error_code)
```

Comment:
I think the critical scopes are only those which might have a `CancelledError` active, and we want to apply this to _all_ (unshielded) `except` blocks.

Reply:
This is just a matter of whether you want both of 102 and 120 to be raised for checkpoints in critical scopes.
You might be getting confused by "except" being a ~magical constant to indicate an except is non-critical? I can add comments to further clarify that, and/or restructure a bunch of things to use a different magical word. The reason for it is it being reused for formatting the error message.